### PR TITLE
propagate link component demo to v1 PL

### PR DESCRIPTION
**Pull request by joekarasek**
This one was pretty straight forward. Added an example of a link with an icon, as well as the icon position options.

### CODE REVIEW

File: apps/pattern-lab--workshop/src/_patterns/02-components/link/10-link-icon-variation.twig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    } only %}
+  </li>
+
+  <li>
+    {% include "@bolt-components-link/link.twig" with {
+      text: "But the icon can go before the link",
+      url: "http://google.com",
+      icon: {
+      name: "add-open",
+        position: "before"
```
Comment (sghoweri) [inline apps/pattern-lab--workshop/src/_patterns/02-components/link/10-link-icon-variation.twig]: Silly nitpick but can we indent 'name' here?

File: packages/components/bolt-link/link.schema.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: Link
+type: object
+properties:
+  url or href:
+    type: string
+    description: Link href attribute. Either url or href may be passed in.
+  icon:
```
Comment (sghoweri) [inline packages/components/bolt-link/link.schema.yml]: A bit of a question for you and @evanlovely - given that the Link component accepts a fully configurable Icon component (which includes additional options like size, color, and background), how best do we document that in the schema?

We want to make sure that as components get updated and schemas evolve, it gets up to date everywhere without a ton of manual work. Thoughts?

File: packages/components/bolt-link/link.schema.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    type: object
+    description: Bolt icon.
+    properties:
+      position:
```
Comment (sghoweri) [inline packages/components/bolt-link/link.schema.yml]: Technically we're missing the name attribute here but per my last question *shrug*

File: packages/components/bolt-link/link.schema.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: Link
+type: object
+properties:
+  url or href:
```
Comment (sghoweri) [inline packages/components/bolt-link/link.schema.yml]: Does this work for a schema prop?!

Comment (joekarasek) [reply packages/components/bolt-link/link.schema.yml]: No, but one can wish. Removing this.

**Comment by joekarasek** [comment]
schema has been disabled and removed from the demo at this time

**Comment by rockymountainhigh1943** [comment]
@sghoweri -- the changes I added to this PR should get us merged and I'd specifically like you to see the prototype I worked up to include schema data from within other components.

In the case of the link component -- we can now pass a reference to the icon schema directly from within the link schema `ref: '@bolt-components-icon/icon.schema.yml'`. This benefits us in:

1. Providing complete documentation for components that include other components
2. Prevents duplication of documentation and allows for a very clean method of including nested component schema data (keeping files separate)
3. The reference added to the included component's schema passes directly into the schema validation!

Take a look at this locally and if you're happy with it merge it on it!

**Pull request merged**